packages/back-end/src/controllers/features.ts

```diff
@@ -1873,7 +1872,27 @@ export async function putFeatureRule(
       });
     }
   }
-
+  let hasChanges = false;
+  const liveRule = feature.environmentSettings[environment].rules.find(
+    (r) => r.id === rule.id
+  );
```

Comment:
I think this is wrong?  What if you are looking at a draft?  I think you'd want to compare against the existing draft to see if there are changes.